tests/cases/fourslash/autoImportAllowImportingTsExtensionsPackageJsonImports1.ts

```diff
@@ -0,0 +1,44 @@
+/// <reference path="fourslash.ts" />
+
+// @module: nodenext
+// @allowImportingTsExtensions: true
+
+// @Filename: /node_modules/pkg/package.json
+//// {
+////   "name": "pkg",
+////   "type": "module",
+////   "exports": {
+////     "./*": {
+////       "types": "./types/*",
+////       "default": "./dist/*"
+////     }
+////   }
+//// }
+
+// @Filename: /node_modules/pkg/types/external.d.ts
+//// export declare function external(name: string): any;
+
+// @Filename: /package.json
+//// {
+////   "name": "self",
+////   "type": "module",
+////   "imports": {
+////     "#*": "./src/*"
+////   },
+////   "exports": {
+////     "./lib/*": "./lib/*"
+////   },
+////   "dependencies": {
+////     "pkg": "*"
+////   }
+//// }
+
+// @Filename: /src/add.ts
+//// export function add(a: number, b: number) {}
+
+// @Filename: /src/index.ts
+//// add/*imports*/;
+//// external/*exports*/;
+
+verify.importFixModuleSpecifiers("imports", ["#add.ts"]);
+verify.importFixModuleSpecifiers("exports", ["pkg/external.js"]);
```

Comment:
Using `.ts` extensions where possible is limited to `"imports"`, and doesn’t take effect for `"exports"`. We may have to revisit this in the coming years if npm fills up with TS-only packages. We do not currently offer auto-imports for self-name imports (which are resolved through your own package.json `"exports"`), which is a place where it might make sense to respect `allowImportingTsExtensions`, so this may need to be revisited in the future as well.